packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -416,15 +423,25 @@ export async function transformGlobImport(
                 throw new Error(
                   "In virtual modules, all globs must start with '/'",
                 )
-              const filePath = `/${relative(root, file)}`
-              return { filePath, importPath: filePath }
+              const importPath = `/${relative(root, file)}`
+              const filePath = options.base
+                ? `${relative(posix.join(root, options.base), file)}`
+                : importPath
+              return { filePath, importPath }
```

Comment:
Should this also handle prepending the `./` if `options.base` is set?

This part applies to virtual modules which don't exist in the file system, so without a `base` we don't know what the relative path would be, so we always made it a root-relative path, like `/src/dir/foo.js`. But if the `base` is set, it could be a way for the virtual module to tell, "resolve it as if I'm in this directory", the `base` would have only allowed to start with `/` though.

Reply:
I see that options.base makes it possible to use relative paths even in the vitual module.
I changed it in [2ec1175](https://github.com/vitejs/vite/pull/18510/commits/2ec1175391ae4d4324870cc43fdbf1fff081a979) because I find it useful.